framework/src/controls/PIDTransientControl.C

```diff
@@ -145,7 +147,12 @@ PIDTransientControl::execute()
     _integral += delta * _dt;
     value += _Kint * _integral + _Kpro * delta;
     if (_dt > 0)
-      value += _Kder * delta / _dt;
+    {
+      if (_reset_every_timestep)
+        value += _Kder * (delta - _old_delta_unaffected) / _dt;
+      else
```

Comment:
If the _reset_every_timestep is true, it is changing the _old_delta to 0 which cannot be used for the derivative term anymore. That's why in this case I have introduced another term _old_delta_affected. The _old_delta is saved in _old_delta_unaffected before resetting it to zero. The _old_delta_unaffected is then used for the calculation of the derivative term. 'If' statement because this only needs to be done if the _old_delta is reset.

Reply:
> If the _reset_every_timestep is true, it is changing the _old_delta to 0 which cannot be used for the derivative term anymore.

on which line is this happening?

Reply:
126, 127

Reply:
it's not though
this
```
    if (_t_step != _t_step_old)
    {
      // Reset the error integral if PID is only used within each timestep
      if (_reset_every_timestep)
        _integral = 0;
      _integral_old = _integral;
      _value_old = value;
      _t_step_old = _t_step;
      _old_delta_unaffected = _old_delta;
      _old_delta = 0;
    }

```
is equivalent to this
```
    if (_t_step != _t_step_old)
    {
      // Reset the error integral if PID is only used within each timestep
      if (_reset_every_timestep)
      {
        _integral = 0;
      }
      _integral_old = _integral;
      _value_old = value;
      _t_step_old = _t_step;
      _old_delta_unaffected = _old_delta;
      _old_delta = 0;
    }
```
so _reset_every_timestep does not affect `_old_delta`

Reply:
not sure
you could just use ` _old_delta_unaffected ` for that line all the time

I m not clear on whether ` _old_delta_unaffected ` is currenttly truly unaffected. I ll have to think more about it

Reply:
alright I'm using _old_delta_unaffected all the time for derivative term now. please check.